cuml/pca_mnmg_demo.ipynb

```diff
@@ -0,0 +1,298 @@
+    "n_features = 20\n",
+    "\n",
+    "n_components = 2\n",
+    "whiten = False\n",
+    "random_state = 32\n",
+    "svd_solver = \"full\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Data\n",
+    "\n",
+    "### GPU"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "X_cudf, _ = make_blobs(n_samples, \n",
+    "                       n_features, \n",
+    "                       centers=1, \n",
+    "                       cluster_std=0.01,\n",
+    "                       random_state=random_state)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Host"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wait(X_cudf)\n",
+    "\n",
+    "X_df = to_dask_df(X_cudf).to_dask_array(lengths=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scikit-learn Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "pca_sk = skPCA(n_components=n_components,\n",
+    "               svd_solver=svd_solver, \n",
+    "               whiten=whiten, \n",
+    "               random_state=random_state)\n",
+    "\n",
+    "result_sk = pca_sk.fit_transform(X_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## cuML Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "pca_cuml = cumlPCA(n_components=n_components,\n",
+    "                   svd_solver=svd_solver, \n",
+    "                   whiten=whiten,\n",
+    "                   random_state=random_state)\n",
+    "\n",
+    "result_cuml = pca_cuml.fit_transform(X_cudf)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluate Results\n",
+    "\n",
+    "### Singular Values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "passed = np.allclose(pca_sk.singular_values_, \n",
+    "                     pca_cuml.singular_values_.to_array(), \n",
+    "                     atol=0.01)\n",
+    "print('compare pca: cuml vs sklearn singular_values_ {}'.format('equal' if passed else 'NOT equal'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Explained Variance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "passed = np.allclose(pca_sk.explained_variance_, \n",
+    "                     pca_cuml.explained_variance_.to_array(), \n",
+    "                     atol=1e-8)\n",
```

Comment:
Woah those tolerances are tight. I think we'd be ok with looser tolerances.

Reply:
I took this from the PCA notebook. The numbers should be pretty exact.